core/src/main/scala/org/apache/spark/input/PortableDataStream.scala

```diff
@@ -47,7 +47,7 @@ private[spark] abstract class StreamFileInputFormat[T]
   def setMinPartitions(sc: SparkContext, context: JobContext, minPartitions: Int) {
     val defaultMaxSplitBytes = sc.getConf.get(config.FILES_MAX_PARTITION_BYTES)
     val openCostInBytes = sc.getConf.get(config.FILES_OPEN_COST_IN_BYTES)
-    val defaultParallelism = sc.defaultParallelism
+    val defaultParallelism = Math.max(sc.defaultParallelism, minPartitions)
```

Comment:
If `sc.defaultParallelism` < 2, and `minParititions` is not set in `BinaryFileRDD`, then previously `defaultParallelism` shall be the same as `sc.defaultParallelism`, and after the change it will be `2`. Have you already consider this case and feel it's right behavior change to make?

Reply:
I metioned `BinaryFileRDD` not this method, you can check the code to see how it handles the default value.

Reply:
BinaryFileRDD will set minPartitions, which will either be defaultMinPartitions, or the values you can set via binaryFiles(path, minPartitions) method. Eventually, this minPartitions value will be passed to setMinPartitions() method.